packages/native/types/index.d.ts

```diff
@@ -0,0 +1,48 @@
+// Definitions by: Pat Sissons <https://github.com/patsissons>
+// TypeScript Version: 3.4
+
+import '@emotion/core'
+
+import {
+  CreateStyled as BaseCreateStyled,
+  CreateStyledComponent,
+  Interpolation,
+  ReactNativeStyle,
+  ReactNativeComponentNames,
+  ReactNativeComponentProps,
+  ReactNativeComponents,
+  ReactNativeStyleType
+} from './base'
+
+export {
+  ArrayInterpolation,
+  CreateStyledComponent,
+  FunctionInterpolation,
+  Interpolation,
+  ObjectInterpolation,
+  ReactNativeStyle,
+  StyledComponent,
+  StyledOptions
+} from './base'
+
+export function css(
+  template: TemplateStringsArray,
+  ...args: Array<Interpolation>
+): ReactNativeStyle
+export function css(...args: Array<Interpolation>): ReactNativeStyle
+
+export type StyledComponents<Theme extends {} = any> = {
+  [ComponentName in ReactNativeComponentNames]: CreateStyledComponent<
+    { theme?: Theme },
+    ReactNativeComponentProps<ComponentName>,
+    { theme: Theme },
+    ReactNativeStyleType<ReactNativeComponentProps<ComponentName>>
```

Comment:
all styled `react-native` components should now be strongly typed against their specific style prop shape.